Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/service/impl/ConceptMergeService.java

```diff
@@ -0,0 +1,235 @@
+
+        String description = conceptMergeBean.getDescriptions().stream().map(i -> i.toString())
+                .collect(Collectors.joining(prefix));
+        entry.setDescription(description);
+
+        String synonymIds = conceptMergeBean.getSynonymsids().stream().map(i -> i.toString())
+                .collect(Collectors.joining(prefix));
+        entry.setSynonymIds(synonymIds);
+
+        String equals = conceptMergeBean.getEqualsValues().stream().map(i -> i.toString())
+                .collect(Collectors.joining(prefix));
+        entry.setEqualTo(equals);
+
+        String similar = conceptMergeBean.getSimilarValues().stream().map(i -> i.toString())
+                .collect(Collectors.joining(prefix));
+        entry.setSimilarTo(similar);
+
+        String wordnetIds = conceptMergeBean.getWordnetIds().stream().map(i -> i.toString())
+                .collect(Collectors.joining(prefix));
+        entry.setWordnetId(wordnetIds);
+
+        addAlternativeIdsToEntry(entry, conceptMergeBean);
+        addMergedIdsToEntry(entry, conceptMergeBean);
+    }
+
+    private void addAlternativeIdsToEntry(ConceptEntry entry, ConceptsMergeBean conceptMergeBean) {
+        List<String> conceptIds = conceptMergeBean.getConceptIds();
+        String selectedId = conceptMergeBean.getSelectedConceptId();
+        Set<String> alternativeIds = new HashSet<>();
+
+        for (String conceptId : conceptIds) {
+            if (!conceptId.equalsIgnoreCase(selectedId) || ConceptTypes.SPECIFIC_WORDNET_CONCEPT == conceptTypesService
+                    .getConceptTypeByConceptId(selectedId)) {
+                alternativeIds.add(conceptId);
+                ConceptEntry conceptEntry = conceptManager.getConceptEntry(conceptId);
+                alternativeIds.addAll(conceptEntry.getAlternativeIds());
+            }
+        }
+        entry.setAlternativeIds(alternativeIds);
+    }
+
+    private Predicate<String> isSpecificWordnetConcept(String selectedId) {
+        return id -> !id.equalsIgnoreCase(selectedId)
+                || ConceptTypes.SPECIFIC_WORDNET_CONCEPT == conceptTypesService.getConceptTypeByConceptId(selectedId);
+    }
+
+    private void addMergedIdsToEntry(ConceptEntry entry, ConceptsMergeBean conceptMergeBean) {
+        String prefix = ",";
+        String selectedId = conceptMergeBean.getSelectedConceptId();
+        String mergedIds = conceptMergeBean.getConceptIds().stream().map(i -> i.toString())
+                .filter(isSpecificWordnetConcept(selectedId))
+                .collect(Collectors.joining(prefix));
+        entry.setMergedIds(mergedIds);
+    }
+
+    @Override
+    public Set<String> getAllConceptLists() {
+        List<ConceptList> allLists = conceptListManager.getAllConceptLists();
+        Set<String> conceptLists = new HashSet<>();
+        for (ConceptList conceptList : allLists) {
+            conceptLists.add(conceptList.getConceptListName());
+        }
+        return conceptLists;
+    }
```

Comment:
why has this its own method? If you can use the list of ConceptLists itself, it should be a one liner in java 8 to turn that into a collection of strings.